Lombok compatibility issue with Java 17 on macOS

**Changes**
- Lombok 1.18.24 -> Lombok 1.18.30

**Additional notes**
Getting compilation error when building with Java 17 on macOS:
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.12.1:compile (default-compile) on project base: Fatal error compiling: java.lang.NoSuchFieldError: Class com.sun.tools.javac.tree.JCTree$JCImport does not have member field 'com.sun.tools.javac.tree.JCTree qualid

**Acceptance Criteria**
- [x] Project builds successfully on macOS with Java 17
- [x] No compilation errors related to Lombok

**Risks? (optional)**
Could affect other developers' local environments!

--- a/pom.xml
+++ b/pom.xml
@@ -197,7 +197,7 @@
     <dependency>
       <groupId>org.projectlombok</groupId>
       <artifactId>lombok</artifactId>
-      <version>1.18.24</version>
+      <version>1.18.30</version>
       <scope>provided</scope>
     </dependency>
     <dependency>

